toolchain/Dockcross.cmake

```diff
@@ -0,0 +1,11 @@
+include($ENV{CMAKE_TOOLCHAIN_FILE})
+
+if($ENV{DEFAULT_DOCKCROSS_IMAGE} MATCHES "dockcross/windows-.*")
+  set(CMAKE_SYSTEM_NAME Windows)
+else()
+  set(CMAKE_SYSTEM_NAME Linux)
+endif()
+
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
```

Comment:
I allow container default toolchain file to load first, but then to iron out some quirks:
- Set the `CMAKE_SYSTEM_NAME` manually, as windows/aarch64 has Linux hardcoded somehow
- Override `CMAKE_FIND_ROOT_PATH` because windows_32 and windows_64 containers have these settings set differently, which causes build to fail - out toolchains all have these like this.

Reply:
I assume Solaris never calls this file?

Reply:
Correct, this file is selected by ant task if `os.target.toolchain` is present. This does not happen in any of the builds for solars or macos.